File: src/main/java/ru/nsu/fit/g20204/kuznetsov/wireframe/model/Geometry.java
package ru.nsu.fit.g20204.kuznetsov.wireframe.model;

import ru.nsu.fit.g20204.kuznetsov.wireframe.math.Vector;

import java.util.ArrayList;
import java.util.List;

public class Geometry {
    private final List<Vector> vectorList;
    private final List<Integer> edgeList;

    public Geometry() {
        vectorList = new ArrayList<>();
        edgeList = new ArrayList<>();
        vectorList.add(new Vector());
    }

    public Geometry(List<Vector> vertexList, List<Integer> edgeList) {
        this.vectorList = vertexList;
        this.edgeList = edgeList;
    }

    public List<Vector> getVertexList() {
        return vectorList;
    }

    public List<Integer> getEdgeList() {
        return edgeList;
    }
}

File: src/main/java/ru/nsu/fit/g20204/kuznetsov/wireframe/model/Geometry.kt
package ru.nsu.fit.g20204.kuznetsov.wireframe.model

import ru.nsu.fit.g20204.kuznetsov.wireframe.math.Vector

class Geometry {
    private val vectorList: MutableList<Vector>
    val edgeList: List<Int>

    constructor() {
        vectorList = ArrayList()
        edgeList = ArrayList()
        vectorList.add(Vector(1.0, 1.0, 1.0, 1.0))
        vectorList.add(Vector(0.0, 0.0, 0.0, 1.0))
    }

    constructor(vertexList: MutableList<Vector>, edgeList: List<Int>) {
        vectorList = vertexList
        this.edgeList = edgeList
    }

    val vertexList: List<Vector>
        get() = vectorList
}
